### PR TITLE
RHDEVDOCS-6127: Content creation for Persistent volumes for Repo server storage

### PR DESCRIPTION
**Pull request by Dhruv-Soni11**
Version(s):

GitOps 1.14

Issue:

https://issues.redhat.com/browse/RHDEVDOCS-6127

Link to docs preview:

[Repo server properties](https://80788--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/argo-cd-cr-component-properties.html)

[Persistent volumes for Repo server storage](https://80788--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/setting-up-argocd-instance.html) hyperlink added under the Additional Resources section

QE review:
- [x] QE has approved this change.

SME review: @svghadi
Peer review: @adellape 
QE review: @varshab1210 

Additional information:

### CODE REVIEW

**Comment by svghadi** [comment]
Hi @Dhruv-Soni11 , any particular reason why the [Repo Server Properties](https://docs.openshift.com/gitops/1.12/argocd_instance/argo-cd-cr-component-properties.html#argo-repo-server-properties_argo-cd-cr-component-properties) is a separate table and not part of the main [Argo CD custom resource properties](https://80788--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/argo-cd-cr-component-properties#argo-cd-properties_argo-cd-cr-component-properties). I think it should be under the main table as a new row.

**Comment by Dhruv-Soni11** [comment]
> Hi @Dhruv-Soni11 , any particular reason why the [Repo Server Properties](https://docs.openshift.com/gitops/1.12/argocd_instance/argo-cd-cr-component-properties.html#argo-repo-server-properties_argo-cd-cr-component-properties) is a separate table and not part of the main [Argo CD custom resource properties](https://80788--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/argo-cd-cr-component-properties#argo-cd-properties_argo-cd-cr-component-properties). I think it should be under the main table as a new row.

I am not sure why it was separated into two different tables @svghadi. Maybe @Srivaralakshmi can shed some light into this.

**Comment by svghadi** [comment]
Another suggestion, can be discussed and made in a separate PR. Adding a index in side for Addition resources. Something like

image

**Comment by Dhruv-Soni11** [comment]
> Another suggestion, can be discussed and made in a separate PR. Adding a index in side for Addition resources. Something like
> 
> image



> Another suggestion, can be discussed and made in a separate PR. Adding a index in side for Addition resources. Something like
> 
> image

Hi @svghadi - I don't think we follow a process where we can additional resources information on the right side of the page. May I know what is the reason behind this ask? CC: @Srivaralakshmi

**Comment by Dhruv-Soni11** [comment]
> Another suggestion, can be discussed and made in a separate PR. Adding a index in side for Addition resources. Something like
> 
> image

Hi @svghadi - I don't think we follow a process where we can additional resources information on the right side of the page. May I know what is the reason behind this ask? CC: @Srivaralakshmi

**Comment by svghadi** [comment]
Personally, I believe it will make it easier for users to navigate and find things.

**Review by svghadi — APPROVED** [review]
/lgtm

File: modules/gitops-repo-server-properties.adoc [hunk 1]
```diff
@@ -23,6 +23,9 @@ The following properties are available for configuring the Repo server component
 |`execTimeout` | `180` |Execution timeout in seconds for rendering tools (for example Helm or Kustomize).
 |`env` | `__<empty>__` |Environment to set for the repository server workloads.
 |`replicas` | `__<empty>__` |The number of replicas for the Argo CD Repo server. Must be greater than or equal to `0`.
+|`volumes` | `__<empty>__` |Configures additional volumes used for the repo server deployment. This field is optional.
+|`volumeMounts` | `__<empty>__` |Configures additional volume mounts used for the repo server deployment. This field is optional.
+|`enabled` | `true` |The flag used to enable the repo server during Argo CD installation.
```
Comment (varshab1210) [inline modules/gitops-repo-server-properties.adoc:28]: There is a known issue with this field
https://issues.redhat.com/browse/GITOPS-3667

Comment (Dhruv-Soni11) [reply modules/gitops-repo-server-properties.adoc:28]: Ok, I have removed the content as per your suggestion. Fixed.

**Comment by varshab1210** [comment]
LGTM, thanks

File: argocd_instance/setting-up-argocd-instance.adoc [hunk 2]
```diff
@@ -24,6 +24,7 @@ include::modules/gitops-argo-cd-installation.adoc[leveloffset=+1]
 * link:https://argocd-operator.readthedocs.io/en/latest/usage/routes/#setting-tls-modes-for-routes[Configuring the route TLS termination]
 * xref:../argocd_instance/argo-cd-cr-component-properties.adoc#argo-cd-properties_argo-cd-cr-component-properties[Argo CD custom resource properties]
 * link:https://docs.openshift.com/container-platform/latest/rest_api/network_apis/route-route-openshift-io-v1.html#spec-tls[Specification for TLS termination configuration]
+* link:https://argocd-operator.readthedocs.io/en/latest/usage/ha/repo-server[Persistent volumes for Repo server storage]
```
Comment (adellape) [inline argocd_instance/setting-up-argocd-instance.adoc:27]: Looking at the target doc, I see "Repo Server" and "repo server", but not "Repo server". Suggest switching it out here for one of those others.

File: modules/gitops-repo-server-properties.adoc [hunk 3]
```diff
@@ -23,6 +23,8 @@ The following properties are available for configuring the Repo server component
 |`execTimeout` | `180` |Execution timeout in seconds for rendering tools (for example Helm or Kustomize).
 |`env` | `__<empty>__` |Environment to set for the repository server workloads.
 |`replicas` | `__<empty>__` |The number of replicas for the Argo CD Repo server. Must be greater than or equal to `0`.
+|`volumes` | `__<empty>__` |Configures additional volumes used for the repo server deployment. This field is optional.
+|`volumeMounts` | `__<empty>__` |Configures additional volume mounts used for the repo server deployment. This field is optional.
```
Comment (adellape) [inline modules/gitops-repo-server-properties.adoc:27]: Similarly, noticing "Repo server" vs "repo server" here (seems split across our openshift-docs repo as well). Only a nit really, but would be better if there was consistency.

File: argocd_instance/setting-up-argocd-instance.adoc (hunk 2 above)
Comment (Dhruv-Soni11) [reply argocd_instance/setting-up-argocd-instance.adoc:27]: Agreed, thanks for the catch. Fixed.

File: modules/gitops-repo-server-properties.adoc (hunk 3 above)
Comment (Dhruv-Soni11) [reply modules/gitops-repo-server-properties.adoc:27]: Fixed.

**Comment by jab-rh** [comment]
@Dhruv-Soni11, looks like this has a merge conflict now. Can you revisit and resolve? Thanks!

**Comment by maxwelldb** [comment]
Removing merge-review-needed label for now. Add it back after needs-rebase is cleared! 👍

**Comment by Dhruv-Soni11** [comment]
> Removing merge-review-needed label for now. Add it back after needs-rebase is cleared! 👍

I have resolved the conflicts and added the label again after rebase. Thanks for the message @maxwelldb.

**Pull request merged**